### PR TITLE
[no-ticket] minor test updates

### PR DESCRIPTION
**Pull request by andy7i**
Ticket: <Link to Jira ticket>
<Put notes here to help reviewer understand this PR>

Couple commits:
- unimplemented mocks may cause false impression of success and subsequent anguish when tests fail to fail
- remove unnecessary extra code?

---

**PR checklist**

- [ ] I've followed [the instructions](https://github.com/broadinstitute/sam/blob/develop/CONTRIBUTING.md#api-changes) if I've made any changes to the API, _especially_ if they're breaking changes
- [ ] I've updated the RC_XXX release ticket with any manual steps required to release this change
- [ ] I've updated the [FISMA documentation](https://github.com/broadinstitute/sam/blob/develop/CONTRIBUTING.md#fisma-documentation-changes) if I've made any security-related changes, including auth, encryption, or auditing

### CODE REVIEW

File: src/test/scala/org/broadinstitute/dsde/workbench/sam/openam/MockAccessPolicyDAO.scala [hunk 1]
```diff
@@ -147,11 +147,11 @@ class MockAccessPolicyDAO(private val policies: mutable.Map[WorkbenchGroupIdenti
     )
   }
 
-  override def getResourceParent(resource: FullyQualifiedResourceId, samRequestContext: SamRequestContext): IO[Option[FullyQualifiedResourceId]] = IO(None)
+  override def getResourceParent(resource: FullyQualifiedResourceId, samRequestContext: SamRequestContext): IO[Option[FullyQualifiedResourceId]] = ???
```
Comment (andy7i) [inline src/test/scala/org/broadinstitute/dsde/workbench/sam/openam/MockAccessPolicyDAO.scala:150]: unimplemented mocks may cause false impression of success and subsequent anguish when tests fail to fail

File: src/test/scala/org/broadinstitute/dsde/workbench/sam/api/ResourceRoutesV1Spec.scala [hunk 2]
```diff
@@ -772,7 +754,7 @@ class ResourceRoutesV1Spec extends FlatSpec with Matchers with ScalatestRouteTes
 
   it should "404 when deleting a resource that exists but can't be seen by the user" in {
     val resourceType = ResourceType(ResourceTypeName("rt"), Set(SamResourceActionPatterns.alterPolicies, SamResourceActionPatterns.readPolicies), Set(ResourceRole(ResourceRoleName("owner"), Set(ResourceAction("run")))), ResourceRoleName("owner"))
-    val samRoutes = createSamRoutes(Map(resourceType.name -> resourceType))
```
Comment (andy7i) [inline src/test/scala/org/broadinstitute/dsde/workbench/sam/api/ResourceRoutesV1Spec.scala:775]: createSamRoutes is only used in this testcase (in this file) and it doesn't seem to be necessary for the test to pass.

**Comment by andy7i** [comment]
jenkins retest -- failures seem unrelated

**Comment by andy7i** [comment]
jenkins retest - leo failure

**Review by marctalbott — APPROVED** [review]

**Review by dvoet — APPROVED** [review]

**Pull request merged**